php/libraries/Utility.class.inc

```diff
@@ -1246,5 +1246,50 @@ class Utility
         }
         return checkdate($dateElement['M'], $dateElement['d'], $dateElement['Y']);
     }
+
+    /**
+     * This function transforms the php.ini notation for numbers (like '2M')
+     * to an integer (2*1024*1024 in this case)
+     *
+     * @param string $sSize PHP notation for memory size.
+     *
+     * @return integer Number of bytes.
+     */
+    function convertPHPSizeToBytes($sSize)
+    {
+        if (is_numeric($sSize) ) {
+            return $sSize;
+        }
+        $sSuffix = substr($sSize, -1);
+        $iValue  = substr($sSize, 0, -1);
+        switch(strtoupper($sSuffix)){
+        case 'P':
+            $iValue *= 1024;
+        case 'T':
+            $iValue *= 1024;
+        case 'G':
+            $iValue *= 1024;
+        case 'M':
+            $iValue *= 1024;
+        case 'K':
+            $iValue *= 1024;
+            break;
+        }
+        return $iValue;
+    }
+
+    /**
+     * This function return the maximum file size that the server can
+     * accept.
+     *
+     * @return integer The minimum between the two settings.
+     */
+    function getMaximumFileUploadSize()
```

Comment:
Xavier's new Utility class functions :  convertPHPSizeToBytes() and getMaximumFileUploadSize() 
These could also be used by the Imaging Uploader in the next release - @MounaSafiHarab @mohadesz @nicolasbrossard